hbase-client/src/main/java/org/apache/hadoop/hbase/client/MetricsConnection.java

```diff
@@ -334,27 +381,48 @@ public Counter newMetric(Class<?> clazz, String name, String scope) {
   MetricsConnection(String scope, Supplier<ThreadPoolExecutor> batchPool,
     Supplier<ThreadPoolExecutor> metaPool) {
     this.scope = scope;
+    this.batchPools.add(batchPool);
```

Comment:
can `MetricsConnection` ever receive non-null Pools? Having this code may make a backport to `branch-2` easier... but it seems like it's no longer relevant in `master`... I wonder if we should be removing the ratios, or just returning a hardcoded 0/0 ratio.

But probably easier to keep as-is for now in this PR, and for the easier backports. It just seems weird to me since it looks like the pools are always null here.

Reply:
You are right, the caller in AsyncConnectionImpl.java always pass a null:
```
this.metrics =
        Optional.of(MetricsConnection.getMetricsConnection(this, () -> null, () -> null));
```

Reply:
I kept this code as I was guessing it might be a place holder for passing non-null pools in future.